lib/lbt/analyzer/XMLTemplateAnalyzer.js

```diff
@@ -304,14 +304,49 @@ class XMLTemplateAnalyzer {
 
 	_analyzeChildren(node) {
 		if ( Array.isArray(node.$$) ) {
-			node.$$.forEach( (child) => this._analyzeNode( child ) );
+			node.$$.forEach( (child) => {
+				// add parentNode reference to child to be able to walk up the parent chain
+				child.$parentNode = node;
+				return this._analyzeNode( child);
+			});
+		}
+	}
+
+	/**
+	 * Walks up the parent chain until fnMatch returns either <code>true</code> or there is no more parent
+	 *
+	 * @param {object} node xml node
+	 * @param {Function} fnMatch matcher function, gets node as argument and returns whether or not the current node matches
+	 * @returns {boolean} <code>true</code> if fnMatch finds a match, <code>false</code> otherwise
+	 * @private
+	 */
+	_findInParentChain(node, fnMatch) {
+		if (fnMatch(node)) {
+			return true;
+		}
+		if (node.$parentNode) {
+			return this._findInParentChain(node.$parentNode, fnMatch);
 		}
+		return false;
 	}
 
 	_analyzeCoreRequire(node) {
 		const coreRequire = getAttributeNS(node, XMLVIEW_CORE_REQUIRE_ATTRIBUTE_NS);
 		let requireContext;
 		if ( coreRequire ) {
+			// expression evaluation
+			if (coreRequire.startsWith("{=")) {
+				// check if parent chain contains a "template" node
+				const fnMatch = (node) => {
+					return node.$ns.uri === TEMPLATING_NAMESPACE;
+				};
+				if (this._findInParentChain(node, fnMatch)) {
```

Comment:
Another solution would be to re-use the `this.conditional` flag, which is already set within `_analyzeNode` when a template tag or its children are processed.

Maybe renaming the flag to some more meaningful name (something that includes templating) would make sense.

Reply:
But `conditional` __is__ the meaning: the node is visited at runtime only _conditionally_. The same property is used throughout the `JSModuleAnalyzer`.

Reply:
I would introduce a new flag `templateTag`, but the idea is good to have a flag which is set just before the children processing and reset afterwards

Reply:
tackled with: 3269bb0

Reply:
@codeworrior conditional is only used for the added dependencies, right?
In this case we have a different use case but based on the same check.

Reply:
conditional is only true for `template:if` and `template:repeat` for others like `template:with` it is false